frontend/src/metabase/dashboard/hoc/DashboardControls.jsx

```diff
@@ -66,7 +67,8 @@ export default ComposedComponent =>
             ? null
             : options.refresh,
         );
-        this.setNightMode(options.theme === "night" || options.night); // DEPRECATED: options.night
+        this.setNightMode(options.theme === "night");
+        this.setTransparent(options.theme === "transparent");
```

Comment:
@ranquild Please see the explanation in the video. Basically, this won't work as I want to capture what `mode` users originally load the dashboard with. If we were to consolidate both flag, then when users switch to the night mode and about to switch back, how could we determine if we want to switch back to `transparent` or `light` mode as we already override the `theme` state. That "original mode" needs to exist somewhere and in this case, it's `isTransparent` state.

https://www.loom.com/share/2f94586840674b389f2e8ceb441b4fdd

Reply:
@WiNloSt Is the behavior you are talking about intentional? I can't find anything about it in the product doc, but it seems to me that we might need to update this light/dark mode toggle to reflect that we have 3 values instead of 2. Could you please clarify this? Thanks!

Reply:
It's not in the doc. I just came across this while implementing and made a judgment on what's the best behavior for this toggle. I'd loop @cdeweyx in to get some opinion. But I think in my opinion this is better than starting off with transparent mode, then toggles to night mode, but when toggling back it goes to light mode instead. I imagine that could feel more like a bug than the behavior I've implemented.

Reply:
Hm I wasn't aware that those options showed up for public embeds. 

Would it be tough to remove the "Nighttime mode" / "Daytime mode" button when the embed background is set to transparent? It feels wrong for me to tweak this within the iFrame if I already said I want it to match the rest of the app (I could just add a global dark mode switch for example).

Reply:
cc: @WiNloSt

Reply:
Can I try remove the nighttime mode toggle altogether no matter what theme is set, so it's more consistent? What do you think?

But to answer your question, yeah I could try that.

Reply:
For visibility. I'll have a go at removing the night/day toggle when embedding with `transparent` theme.